src/path.ts

```diff
@@ -0,0 +1,187 @@
+ */
+export function fromPosix(path: string): DistinctivePath {
+  const split = path.replace(/^\/+/, "").split("/")
+  if (path.endsWith("/")) return { directory: split.slice(0, -1) }
+  return { file: split }
+}
+
+/**
+ * Transform a `DistinctivePath` into a string.
+ *
+ * Directories will have the format `path/to/dir/` and
+ * files will have the format `path/to/file`.
+ */
+export function toPosix(
+  path: DistinctivePath,
+  { absolute }: { absolute: boolean } = { absolute: false }
+): string {
+  const prefix = absolute ? "/" : ""
+  const joinedPath = unwrap(path).join("/")
+  if (isDirectory(path)) return prefix + joinedPath + (joinedPath.length ? "/" : "")
+  return prefix + joinedPath
+}
+
+
+
+// 🛠
+
+
+/**
+ * Combine two `DistinctivePath`s.
+ */
+export function combine(a: DirectoryPath, b: DistinctivePath): DistinctivePath {
+  return map(p => unwrap(a).concat(p), b)
+}
+
+/**
+ * Is this `DistinctivePath` of the given `Branch`?
+ */
+export function isBranch(branch: Branch, path: DistinctivePath): boolean {
+  return unwrap(path)[0] === branch
+}
+
+/**
+ * Is this `DistinctivePath` a directory?
+ */
+export function isDirectory(path: DistinctivePath): path is DirectoryPath {
+  return !!(path as DirectoryPath).directory
+}
+
+/**
+ * Is this `DistinctivePath` a file?
+ */
+export function isFile(path: DistinctivePath): path is FilePath {
+  return !!(path as FilePath).file
+}
+
+/**
+ * Is this `DirectoryPath` a root directory?
+ */
+export function isRootDirectory(path: DirectoryPath): boolean {
+  return path.directory.length === 0
+}
+
+/**
+ * Check if two `DistinctivePath` have the same `Branch`.
+ */
+export function isSameBranch(a: DistinctivePath, b: DistinctivePath): boolean {
+  return unwrap(a)[0] === unwrap(b)[0]
+}
+
+/**
+ * Map a `DistinctivePath`.
+ */
+export function map(fn: (p: Path) => Path, path: DistinctivePath): DistinctivePath {
+  if (isDirectory(path)) return { directory: fn(path.directory) }
+  else if (isFile(path)) return { file: fn(path.file) }
+  return path
+}
+
+/**
+ * Get the parent directory of a `DistinctivePath`.
+ */
+export function parent(path: DistinctivePath): Maybe<DirectoryPath> {
+  return isDirectory(path) && isRootDirectory(path as DirectoryPath)
+    ? null
+    : directory(...unwrap(path).slice(0, -1))
+}
+
+/**
+ * Remove the `Branch` of a `DistinctivePath` (ie. the top-level directory)
+ */
+export function removeBranch(path: DistinctivePath): DistinctivePath {
+  return map(
+    p => isDirectory(path) || p.length > 1 ? p.slice(1) : p,
+    path
+  )
+}
+
+/**
+ * Unwrap a `DistinctivePath`.
+ */
+export function unwrap(path: DistinctivePath): Path {
+  if (isDirectory(path)) {
+    return path.directory
+  } else if (isFile(path)) {
+    return path.file
+  }
+
+  return []
+}
+
+
+
+// ⚗️
+
+
+/**
+ * Render a raw `Path` to a string for logging purposes.
+ */
+export function log(path: Path): string {
+  return `[ ${path.join(", ")} ]`
```

Comment:
True. I don't think we need the quotes, because the error is always presented as a path. And since paths can only be a list of strings, I think it doesn't matter.